src/sentry/workflow_engine/types.py

```diff
@@ -32,17 +33,19 @@ class DetectorPriorityLevel(IntEnum):
 ProcessedDataConditionResult = tuple[bool, list[DataConditionResult]]
 
 
-class EventJob(TypedDict):
+@dataclass(frozen=True)
+class EventJob:
     event: GroupEvent
 
 
-class WorkflowJob(EventJob, total=False):
-    group_state: GroupState
-    is_reprocessed: bool
-    has_reappeared: bool
-    has_alert: bool
-    has_escalated: bool
-    workflow: Workflow
+@dataclass(frozen=True)
+class WorkflowJob(EventJob):
```

Comment:
I feel like the naming with `Job` is still tying the data model to the idea of this happening in post_process. could we just name this like `WorkflowEventData` or something along those lines?